utils_clip.py

```
@@ -1,13 +1,80 @@
+            embeddings.append(cache_clip[text])
+        else:
+            # 缓存未命中，记录需要调用API的文本
+            embeddings.append(None)  # 占位符
+            texts_to_embed.append(text)
+            indices_to_embed.append(i)
+    
+    # 如果有需要调用API的文本，批量调用OpenAI API
+    if texts_to_embed:
+        try:
+            # 调用OpenAI Embeddings API
+            response = openai_client.embeddings.create(
```

Ruff (0.12.2)

Undefined name `openai_client`

(F821)